redux/reducersComp/cryptosSlice.ts

```diff
@@ -0,0 +1,85 @@
+import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
+import type {PayloadAction} from '@reduxjs/toolkit';
+import axios from 'axios';
+
+import type {RootState} from '../store';
+import {initialStateProps, DataCriptoInfo} from '../../utils/interfaces';
+
+const initialState: initialStateProps = {
+  cryptosData: [],
+  error: '',
+};
+
+export const cryptoApiData = createAsyncThunk(
+  'crypto/api',
+  async (name: string) => {
+    const response = await axios.get(
+      `https://data.messari.io/api/v1/assets/${name}/metrics`,
+    );
+    if (response.data.data) {
+      const symbolLower = response.data.data.symbol.toLowerCase();
+      return {
+        id: response.data.data.id,
+        name: response.data.data.name,
+        symbol: response.data.data.symbol,
+        price: response.data.data.market_data.price_usd,
+        percentage:
+          response.data.data.market_data.percent_change_usd_last_24_hours,
+        img: {
+          uri: `https://cryptoicons.org/api/color/${symbolLower}/50`,
+        },
+      };
+    } else {
+      return response.data.status;
+    }
+  },
+);
+
+const cryptosSlice = createSlice({
+  name: 'cryptos',
+  initialState,
+  reducers: {
+    addCrypto: (state, action: PayloadAction<DataCriptoInfo | null>) => {
```

Comment:
are you using this reducer?

Reply:
Im using it but in card 4, i will deleted it for this card.